test/integration/redisfailover/creation_test.go

```diff
@@ -0,0 +1,203 @@
+	"github.com/stretchr/testify/require"
+	corev1 "k8s.io/api/core/v1"
+	apiextensionsclientset "k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/client-go/kubernetes"
+	_ "k8s.io/client-go/plugin/pkg/client/auth/oidc"
+	"k8s.io/client-go/util/homedir"
+
+	redisfailoverv1alpha2 "github.com/spotahome/redis-operator/api/redisfailover/v1alpha2"
+	redisfailoverclientset "github.com/spotahome/redis-operator/client/k8s/clientset/versioned"
+	"github.com/spotahome/redis-operator/cmd/utils"
+	"github.com/spotahome/redis-operator/log"
+	"github.com/spotahome/redis-operator/metrics"
+	"github.com/spotahome/redis-operator/operator/redisfailover"
+	"github.com/spotahome/redis-operator/service/k8s"
+	"github.com/spotahome/redis-operator/service/redis"
+)
+
+const (
+	name         = "testing"
+	namespace    = "rf-integration-tests"
+	redisSize    = int32(3)
+	sentinelSize = int32(3)
+)
+
+type clients struct {
+	k8sClient   kubernetes.Interface
+	rfClient    redisfailoverclientset.Interface
+	aeClient    apiextensionsclientset.Interface
+	redisClient redis.Client
+}
+
+func (c *clients) prepareNS() {
+	ns := &corev1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: namespace,
+		},
+	}
+	c.k8sClient.CoreV1().Namespaces().Create(ns)
+}
+
+func (c *clients) cleanup(stopC chan struct{}) {
+	c.k8sClient.CoreV1().Namespaces().Delete(namespace, &metav1.DeleteOptions{})
+	close(stopC)
+}
+
+// Run execs the program.
+func TestRedisFailover(t *testing.T) {
+	require := require.New(t)
+
+	// Create signal channels.
+	stopC := make(chan struct{})
+	errC := make(chan error)
+
+	flags := &utils.CMDFlags{
+		KubeConfig:  filepath.Join(homedir.HomeDir(), ".kube", "config"),
+		Development: true,
+	}
+
+	// Kubernetes clients.
+	stdclient, customclient, aeClientset, err := utils.CreateKubernetesClients(flags)
+	require.NoError(err)
+
+	// Create the redis clients
+	redisClient := redis.New()
+
+	clients := clients{
+		k8sClient:   stdclient,
+		rfClient:    customclient,
+		aeClient:    aeClientset,
+		redisClient: redisClient,
+	}
+
+	// Create kubernetes service.
+	k8sservice := k8s.New(stdclient, customclient, aeClientset, log.Dummy)
+
+	// Prepare namespace
+	clients.prepareNS()
+
+	// Create operator and run.
+	redisfailoverOperator := redisfailover.New(redisfailover.Config{}, k8sservice, redisClient, metrics.Dummy, log.Dummy)
+	go func() {
+		errC <- redisfailoverOperator.Run(stopC)
+	}()
+
+	// Prepare cleanup for when the test ends
+	defer clients.cleanup(stopC)
+
+	// Give time to the operator to start
+	time.Sleep(15 * time.Second)
+
+	ok := t.Run("Check Custom Resource Creation", clients.testCRCreation)
+	require.True(ok, "the custom resource has to be created to continue")
+
+	// Giving time to the operator to create the resources
+	time.Sleep(6 * time.Minute)
+
+	t.Run("Check Redis Statefulset existing and size", clients.testRedisStatefulSet)
```

Comment:
Integration tests are more complex than unit tests, in a unit test you should only test one thing that's why they are self descriptive most of the times, this doesn't happen on integration tests so a comment of what does it helps a lot :)